gateway/src/main/java/com/mx/path/gateway/configuration/ConfigurationBinder.java

```diff
@@ -273,7 +273,7 @@ private void validate(Object obj) {
     List<Annotations.AnnotatedField<ConfigurationField>> annotatedFields = Annotations.fieldsWithAnnotation(ConfigurationField.class, obj.getClass());
     annotatedFields.forEach(annotatedField -> {
       Object value = Fields.getFieldValue(annotatedField.getField(), obj);
-      state.withField(annotatedField.getField().getName(), () -> validateField(annotatedField, value));
+      state.withField(annotatedField.getAnnotation().value(), () -> validateField(annotatedField, value));
```

Comment:
We might need to see if the annotation's value is blank. If it is we want to fall back the field name.

Reply:
Updated that. I've been in Kotlin for a bit and haven't used Java 17, so let me know if there is a more elegant way to do that null check.

Reply:
You can use `Strings.isBlank()` from `com.mx.common.lang` ([source](https://github.com/mxenabled/path-core/blob/master/common/src/main/java/com/mx/common/lang/Strings.java)). The only difference from your code is that it will return false if the string contains only whitespace (which seems reasonable).

Side note: `string.isBlank()` is the Java 11+ equivalent to `string.trim().isEmpty()`, but `com.mx.common.lang.Strings.isBlank()` was implemented before this library was upgraded from Java 8.

Reply:
Ah, much cleaner. Thanks. Updated the commit with that change.